styles/_helpers.scss

```diff
@@ -4,5 +4,5 @@
 }
 
 .is-hidden {
-  display: none;
+  display: none !important;
```

Comment:
The reason is css selector specificity: because `.c-nav > .app-nav` is more specific (two classes instead of one) than `.is-hidden`, it will override it.
Read: https://css-tricks.com/specifics-on-css-specificity/
!important is considered as bad practice, but in that case that is the least inelegant way I think. It makes sense because you want is-hidden to work in every possible case.